src/IECore/TriangulateOp.cpp

```diff
@@ -302,29 +302,34 @@ struct TriangulateOp::TriangulateFn
 		TriangleDataRemap uniformRemap( uniformIndices );
 		for ( PrimitiveVariableMap::iterator it = m_mesh->variables.begin(); it != m_mesh->variables.end(); ++it )
 		{
+			TriangleDataRemap *remap = nullptr;
 			if ( it->second.interpolation == PrimitiveVariable::FaceVarying )
 			{
- 				assert( it->second.data );
-				varyingRemap.m_other = it->second.data.get();
-				DataPtr data = it->second.data->copy();
-
-				size_t primVarSize = despatchTypedData<TriangleDataRemap, TypeTraits::IsVectorTypedData>( data.get(), varyingRemap );
-				assert( primVarSize == faceVaryingIndices.size() );
-				(void)primVarSize;
-
-				it->second.data = data;
+				remap = &varyingRemap;
 			}
 			else if ( it->second.interpolation == PrimitiveVariable::Uniform )
 			{
- 				assert( it->second.data );
-				uniformRemap.m_other = it->second.data.get();
-				DataPtr data = it->second.data->copy();
+				remap = &uniformRemap;
+			}
+			else
+			{
+				continue;
+			}
 
-				size_t primVarSize = despatchTypedData<TriangleDataRemap, TypeTraits::IsVectorTypedData>( data.get(), uniformRemap );
-				assert( primVarSize == uniformIndices.size() );
-				(void)primVarSize;
+			const Data *inputData = it->second.indices ? it->second.indices.get() : it->second.data.get();
+			DataPtr result = inputData->copy();
+			remap->m_other = inputData;
 
-				it->second.data = data;
+			size_t primVarSize = despatchTypedData<TriangleDataRemap, TypeTraits::IsVectorTypedData>( result.get(), *remap );
+			(void)primVarSize;
```

Comment:
Seems odd - if we don't need `primVarSize` then we don't need to create the variable at all do we? I think this was necessary before because `primVarSize` was being used in an assertion, right, but that's now removed?

Reply:
Yeah, I had no idea why that was there so I just left it...

Reply:
Addressed in 0a29856fc7a87ac11080b2bca558d5e71854b237